### PR TITLE
663 - Remove extra search icon in dropdown

### LINKED ISSUE #663: Dropdown with Icon: There's an extra Search icon at the bottom when searching
**Describe the bug**
[All Browsers] Dropdown with Icon: There's an extra Search icon at the bottom when searching

**To Reproduce**
1. Go to http://master-enterprise.demo.design.infor.com/components/dropdown/example-icons.html
2. Click on the Dropdown field, then clear the content
3. Type "Icon with" in the Search field
4. Notice the extra Search icon at the bottom of the Dropdown

**Expected behavior**
Extra Search icon should not be displayed

**Screenshots**
screen shot 2018-08-24 at 5 31 30 pm

**Platform**
 - All browsers in Mac and Windows

**Additional context**
No additional context.

### PR DESCRIPTION
**Pull request by tjamesallen15**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Remove unnecessary search icon

**Related github/jira issue (required)**:

Closes #663 

**Steps necessary to review your pull request (required)**:

http://localhost:4000/components/dropdown/example-icons
type in something
no search icon should be shown on the bottom.

### CODE REVIEW

**Comment by tmcconechy** [comment]
screen shot 2018-08-30 at 1 56 00 pm

1. The text inside the field when typing has a space up where the icon would be.
2. There is also a dropdown at the bottom (where the search was on the bug)

**Comment by EdwardCoyle** [comment]
@tmcconechy the space is there because there actually should be an icon there.  The extra dropdown arrow is due to bad targeting.  It's all covered in the comments on #663

**Comment by tjamesallen15** [comment]
Thanks for all the comments. I will rework on this one.

**Comment by tjamesallen15** [comment]
Hi Team,

Would you check my fix? I read the requirements on the issue page.

Thanks.

**Comment by tjamesallen15** [comment]
This build fails due functional:ci test but no errors when tried in my local.

**Comment by tmcconechy** [comment]
@tjamesallen15 I should have fixed this on https://github.com/infor-design/enterprise/pull/720
So once this is merged you just need to merge master onto your branch with

```
git pull
git merge origin master
```

Should work once thats merged.

**Comment by tjamesallen15** [comment]
Hi @tmcconechy,

The build still fails in e2e:ci but it's not part of my code change.

Thanks.

**Comment by tjamesallen15** [comment]
error still occurs.

**Comment by clepore** [comment]
I fixed the one, but now this test is failing, also dealing with a dropdown and finding the new value.
https://travis-ci.com/infor-design/enterprise/jobs/144053169#L2097

File: src/components/dropdown/dropdown.js [hunk 1]
```diff
@@ -986,11 +988,6 @@ Dropdown.prototype = {
         self.highlightOption(li);
         selected = true;
       }
-
-      // Highlight Term
-      const exp = self.getSearchRegex(term);
-      const text = li.text().replace(exp, '<i>$1</i>').trim();
-      li.removeClass('hidden').children('a').html(text);
```
Comment (clepore) [inline src/components/dropdown/dropdown.js]: I do not think we should remove these lines of code for highlighting the term.

Comment (tjamesallen15) [reply src/components/dropdown/dropdown.js]: @clepore May I know how do these line of codes function? As I have observed, those line of codes removes the svg icons beside the list items. Maybe, I'm just missing something.

**Comment by tmcconechy** [comment]
1. type f -> wrong icon is shown. I suggest we just show a placeholder icon here like the grey circle from http://localhost:4000/components/alerts/example-index.html
screen shot 2018-09-10 at 10 49 07 am

2. if i backspace and clear all values a stray arrow icon is shown.
screen shot 2018-09-10 at 10 50 02 am

**Comment by tmcconechy** [comment]
Any updates @tjamesallen15 or do you need a hand here?

**Comment by tmcconechy** [comment]
Try to merge master - it should fix the tests.

**Comment by tmcconechy** [comment]
While you are still fixing this. I think it would be better to do this...

https://github.com/infor-design/enterprise/pull/713/files#diff-197080e688816258787520857120d94aR1016

By making a css class for it and toggling that.

**Comment by tmcconechy** [comment]
I merged this branch with master. But i am still seeing the issue that @davidcarlsonberg mentioned.

http://localhost:4000/components/dropdown/example-icons
- type a
- type backspace

(see the arrow on the bottom of the page)

**Comment by tjamesallen15** [comment]
Remove the icon below the dropdown list.

File: src/components/dropdown/dropdown.js [hunk 2]
```diff
@@ -1002,6 +1010,14 @@ Dropdown.prototype = {
 
     term = '';
     this.position();
+
+    if (this.list.find('svg').length > 2) {
```
Comment (tmcconechy) [inline src/components/dropdown/dropdown.js]: Suspect this line is causing the mentioned issue

File: src/components/dropdown/dropdown.js [hunk 3]
```diff
@@ -1054,6 +1070,14 @@ Dropdown.prototype = {
 
     lis.removeClass('hidden');
     this.position();
+
+    if (this.list.find('svg').length === 2) {
```
Comment (tmcconechy) [inline src/components/dropdown/dropdown.js]: And this

**Comment by tmcconechy** [comment]
Just one issue left i can see:

- type s
- backspace twice (all the icons go away)

**Comment by davidcarlsonberg** [comment]
I see Tim's and another:
- click label to focus
- type i
- type s
- notice inputted text ('is') appears overlapping the icon

**Comment by EdwardCoyle** [comment]
There's still something converting the icon back to a Circle during filtering...

oct 1 2018 11_07 am

**Pull request merged**